ext/openssl/tests/openssl_csr_export_to_file_basic.phpt

```diff
@@ -0,0 +1,55 @@
+--TEST--
+openssl_csr_export_to_file() tests
+--SKIPIF--
+<?php if (!extension_loaded("openssl")) print "skip"; ?>
+--FILE--
+<?php
+$csrfile = dirname(__FILE__) . "/openssl_csr_export_to_file_csr.tmp";
+$wrong = "wrong";
+$config = __DIR__ . DIRECTORY_SEPARATOR . 'openssl.cnf';
+$config_arg = array('config' => $config);
+
+$dn = array(
+	"countryName" => "BR",
+	"stateOrProvinceName" => "Rio Grande do Sul",
+	"localityName" => "Porto Alegre",
+	"commonName" => "Henrique do N. Angelo",
+	"emailAddress" => "hnangelo@php.net"
+);
+
+$args = array(
+	"digest_alg" => "sha1",
+	"private_key_bits" => 2048,
+	"private_key_type" => OPENSSL_KEYTYPE_DSA,
+	"encrypt_key" => true,
+	"config" => $config,
+);
+
+$privkey = openssl_pkey_new($config_arg);
+$csr = openssl_csr_new($dn, $privkey, $args);
+var_dump(openssl_csr_export_to_file($csr, $csrfile));
+var_dump(openssl_csr_export_to_file($wrong, $csrfile));
+var_dump(openssl_csr_export_to_file($privkey, $csrfile));
+var_dump(openssl_csr_export_to_file(array(), $csrfile));
+var_dump(openssl_csr_export_to_file($csr, $csrfile, false));
+?>
+--CLEAN--
+<?php
+if (file_exists($csrfile)) {
+	unlink($csrfile);
+}
+?>
+--EXPECTF--
+bool(true)
```

Comment:
btw it's kind of confusing and confused me too because if the path is invalid, it doesn't warn or anything and just use default priv key. I think we should probably consider it as a bug.